stdlib/public/Concurrency/Actor.cpp

```diff
@@ -207,6 +207,53 @@ class ActiveTask {
   }
 };
 
+/// A helper class for a defensive retain pattern where you may need to retain
+/// an object in some circumstances. This helper makes it easy to retain the
+/// object only once even when loops are involved, and do a release to balance
+/// the retain on all paths out of the scope.
+class DefensiveRetainHelper {
```

Comment:
Should this be in HeapObject or some other more general purpose file instead of in Actor.cpp?

Reply:
That probably makes sense. `include/swift/Runtime/HeapObject.h` seems appropriate for it. It actually already has a `SwiftRAII` class for doing swift_retain/release with RAII (but not with the weird only-once-maybe-never thing we want here) so this would fit well next to that.

Reply:
Pushed that, renamed to `SwiftDefensiveRetainRAII` to match `SwiftRAII` and moved to `HeapObject.h`. Interestingly, `SwiftRAII` appears to be dead code.